bitmovin-analytics-datasource/src/components/FilterRow.tsx

```diff
@@ -1,136 +1,135 @@
-    return difference(SELECTABLE_QUERY_ATTRIBUTES, selectedAttributes);
+    return differenceWith(
+      SELECTABLE_QUERY_ATTRIBUTES,
+      filters,
+      (selectableValue, selectedValue) => selectableValue.value === selectedValue.attribute
+    );
   }
 };
 
-const mapFiltersToQueryFilters = (filters: Filter[]): QueryFilter[] => {
+const mapFilterRowsToQueryFilters = (filters: FilterRowData[]): QueryFilter[] => {
   return filters.map((filter) => {
     return {
-      name: filter.selectedAttribute.value!,
-      operator: filter.selectedOperator.value!,
+      name: filter.attribute,
+      operator: filter.operator,
       value: filter.convertedFilterValue,
     } as QueryFilter;
   });
 };
 
 type Props = {
   readonly isAdAnalytics: boolean;
-  readonly onChange: (newFilters: QueryFilter[]) => void;
+  readonly onQueryFilterChange: (newFilters: QueryFilter[]) => void;
+  readonly onFilterRowChange: (newFilters: FilterRowData[]) => void;
+  readonly filters: FilterRowData[];
 };
 
 export function FilterRow(props: Props) {
-  const [filters, setFilters] = useState<Filter[]>([]);
-
   const addFilterInput = () => {
-    setFilters((prevState) => [
-      ...prevState,
-      {
-        selectedAttribute: {},
-        selectedOperator: {},
-        rawFilterValue: '',
-        convertedFilterValue: '',
-        parsingValueError: '',
-      } as Filter,
-    ]);
+    const newFilters = [...props.filters];
+    newFilters.push({
+      attribute: {},
+      operator: {},
+      rawFilterValue: '',
+      convertedFilterValue: '',
+      parsingValueError: '',
+    } as FilterRowData);
+    props.onFilterRowChange(newFilters);
   };
 
   const onAddFilter = (index: number) => {
-    const filter = filters[index];
+    const filter = props.filters[index];
     try {
       const convertedValue = convertFilterValueToProperType(
         filter.rawFilterValue,
-        filter.selectedAttribute.value!,
-        filter.selectedAttribute.label!,
-        filter.selectedOperator.value!,
+        filter.attribute,
+        filter.attribute,
+        filter.operator,
         props.isAdAnalytics
       );
 
-      const newFilter = { ...filter, convertedFilterValue: convertedValue, parsingValueError: '' } as Filter;
+      const newFilter = { ...filter, convertedFilterValue: convertedValue, parsingValueError: '' } as FilterRowData;
 
-      const newFilters = [...filters];
+      const newFilters = [...props.filters];
       newFilters.splice(index, 1, newFilter);
 
-      setFilters(newFilters);
+      props.onFilterRowChange(newFilters);
 
-      props.onChange(mapFiltersToQueryFilters(newFilters));
+      props.onQueryFilterChange(mapFilterRowsToQueryFilters(newFilters));
     } catch (e: unknown) {
       if (e instanceof Error) {
         const errorMessage = e.message;
-        const newFilter = { ...filter, parsingValueError: errorMessage } as Filter;
+        const newFilter = { ...filter, parsingValueError: errorMessage } as FilterRowData;
```

Comment:
remove not necessary castings like this `as FilterRowData`.

use instead `const newFilter: FilterRowData = { ...filter, ......`